### PR TITLE
Final Updates During Epoch Processing

### PR DESCRIPTION
**Pull request by terenc3t**
This is part of #1449 

Key changes:

- Finished what's missing during epoch processing's final updates:
```
1.) Set state.latest_index_roots[(next_epoch + ENTRY_EXIT_DELAY) % LATEST_INDEX_ROOTS_LENGTH] = 
hash_tree_root(get_active_validator_indices(state, next_epoch + ENTRY_EXIT_DELAY)).
2.) Set state.latest_penalized_balances[(next_epoch) % LATEST_PENALIZED_EXIT_LENGTH] =
state.latest_penalized_balances[current_epoch % LATEST_PENALIZED_EXIT_LENGTH].
3.) Set state.latest_randao_mixes[next_epoch % LATEST_RANDAO_MIXES_LENGTH] =
get_randao_mix(state, current_epoch).
```
- Tests for the above
- Updated state transition

### CODE REVIEW

File: beacon-chain/core/epoch/epoch_processing.go [hunk 1]
```diff
@@ -263,7 +264,7 @@ func ProcessValidatorRegistry(
 	}
 	mix, err := helpers.RandaoMix(state, randaoMixSlot)
 	if err != nil {
-		return nil, fmt.Errorf("could not get randao mix: %v", err)
+		return nil, fmt.Errorf("could not get seed mix: %v", err)
```
Comment (prestonvanloon) [inline beacon-chain/core/epoch/epoch_processing.go]: Why is this renamed if the method is still randaoMix?

**Comment by rauljordan** [comment]
The spec still calls it get randao mix and it is quite an important helper, so I vote to keep it as randao mix

File: beacon-chain/core/epoch/epoch_processing.go (hunk 1 above)
Comment (terenc3t) [reply beacon-chain/core/epoch/epoch_processing.go]: More I think, randao is better name than seed, randao implies how the randomness gets generated where seed is just implies the source for randomness.

Changing back got randao
